adapters/invibes/invibes.go

```diff
@@ -0,0 +1,327 @@
+package invibes
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+)
+
+const adapterVersion = "prebid_1.0.0"
+
+type InvibesAdRequest struct {
+	BidParamsJson string
+	Location      string
+	Lid           string
+	IsTestBid     bool
+	Kw            string
+	IsAmp         bool
+	Width         string
+	Height        string
+	GdprConsent   string
+	Gdpr          bool
+	Bvid          string
+	InvibBVLog    bool
+	VideoAdDebug  bool
+}
+type InvibesBidParams struct {
+	PlacementIds []string
+	BidVersion   string
+	Properties   map[string]InvibesPlacementProperty
+}
+type InvibesPlacementProperty struct {
+	Formats []openrtb.Format
+	ImpId   string
+}
+type InvibesInternalParams struct {
+	BidParams   InvibesBidParams
+	Host        string
+	IsAmp       bool
+	Gdpr        string
+	GdprConsent string
+
+	TestIp   string
+	TestBvid string
+	TestLog  bool
+}
+type BidServerBidderResponse struct {
+	Currency  string              `json:"currency"`
+	TypedBids []BidServerTypedBid `json:"typedBids"`
+	Error     string              `json:"error"`
+}
+type BidServerTypedBid struct {
+	Bid          openrtb.Bid `json:"bid"`
+	DealPriority int         `json:"dealPriority"`
+}
+
+func (a *InvibesInternalParams) IsTestRequest() bool {
+	return a.TestIp != "" || a.TestBvid != ""
+}
+
+type InvibesAdapter struct {
+	EndpointTemplate template.Template
+}
+
+func NewInvibesBidder(endpointTemplate string) *InvibesAdapter {
+	urlTemplate, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unable to parse endpoint url template")
+		return nil
+	}
+	return &InvibesAdapter{EndpointTemplate: *urlTemplate}
+}
+
+func (a *InvibesAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
```

Comment:
I know other bidders have this check. I don't believe it's necessary though. The core code will not call `MakeRequests` if there are no impressions. Given this information, I'll trust whatever choice you make here.